### DISCUSSION TITLE
Manually trigger useContractRead refresh

### DISCUSSION THREAD

**Discussion by montanaflynn** [wevm/wagmi#1387] [Q&A] [answered]
I have a `useContractRead` that I would like to manually trigger after a `useContractWrite`, I don't want to use the `watch` since I've found it to be unreliable and create too many unnecessary calls. 

Is it possible to trigger the `useContractRead` in some other way? Similar to the `useContractWrite`'s `write` function or in another way?

Thanks.

**Comment by tmm** [comment, marked as answer]
Yup! Just use [`refetch`](https://wagmi.sh/react/hooks/useContractRead#return-value):

```ts
const { refetch } = useContractRead({ … })
```

Depending on what you are trying to do, you might also want to look into [`useWaitForTransaction`](https://wagmi.sh/react/hooks/useWaitForTransaction).

**Comment by montanaflynn** [comment]
Thank you, this is exactly what I was looking for! Not sure how I missed it tbh 🤦

**Reply by 0xCardiE** [reply]
Great used it as

` const { data: totalSupplyData, refetch: refetchSupply } = useContractRead({`

should be in docs as an example.
